### PR TITLE
UI: Add necessary params to links in Access & KV v2

### PR DESCRIPTION
**Pull request by hashishaw**
**Note to reviewer** This will likely be easiest to review if you look at the changes commit-by-commit. I grouped changes that relate to each other together and that way you won't have to switch contexts as you might by just looking at all the files changed. 

As part of the work to handle Ember Deprecations for 5+, I found that our test suite starting failing spectacularly when replacing `this.replaceWith` in [logout.js](https://github.com/hashicorp/vault/blob/main/ui/app/routes/vault/cluster/logout.js) with `this.router.replaceWith`. The error looked something like: 
```
TypeError: While generating link to route "route.name": Cannot read properties of undefined (reading 'shouldSupersede')
    at NamedTransitionIntent.applyToHandlers
```
There are no issues in the app itself because the offending line of code is only run in the test suite, but when I instead used `this.router.replaceWith` for non-test builds I was able to see the error in the UI. So, rather than doing a bandaid fix such as always navigating to the dashboard page before logging out, I thought the more stable thing to do is to fix the root of the problem, which is add the missing route params on the links.  

- [x] Ent tests passed

### CODE REVIEW

File: ui/lib/kv/addon/components/page/secret/metadata/version-history.hbs [hunk 1]
```diff
@@ -77,15 +94,18 @@
                 @hasChevron={{false}}
                 data-test-popup-menu-trigger
               />
+              {{! TODO: replace }}
```
Comment (hellobontempo) [inline ui/lib/kv/addon/components/page/secret/metadata/version-history.hbs:97]: Is this TODO going to be tackled in a follow-on PR? If so, could the comment be more descriptive?

File: ui/lib/kv/addon/utils/kv-breadcrumbs.js [hunk 2]
```diff
@@ -32,8 +32,8 @@ function splitSegments(secretPath) {
  * @param {boolean} lastItemCurrent
  * @returns array of breadcrumbs specific to KV engine
  */
-export function breadcrumbsForSecret(secretPath, lastItemCurrent = false) {
-  if (!secretPath) return [];
+export function breadcrumbsForSecret(backend, secretPath, lastItemCurrent = false) {
```
Comment (hellobontempo) [inline ui/lib/kv/addon/utils/kv-breadcrumbs.js:35]: It'd be helpful if these args were named args, but I understand that's likely outside the scope of this PR

**Review by hellobontempo — APPROVED** [review]
Great work! Thanks for the detailed PR description and clean commits! It made reviewing much easier. Just a question about a TODO comment

**Review by noelledaley — APPROVED** [review]
this must've been so tricky to hone in on. nice job debugging + solving this! 🙌

File: ui/lib/kv/addon/components/page/secret/metadata/version-history.hbs (hunk 1 above)
Comment (hashishaw) [reply ui/lib/kv/addon/components/page/secret/metadata/version-history.hbs:97]: Oh, that was my note to self to change the link below, which I did. Will remove

**Pull request merged**